### ISSUE TITLE
Missing brain for numpy.dtypes (e.g. StringDType)

### ISSUE THREAD

**Issue by Julfried** [pylint-dev/astroid#3289]
### Steps to reproduce

1. Install any NumPy with the `dtypes` module (1.25 or later, including 2.x).
2. Run:
```python
from astroid import extract_node
node = extract_node("""
import numpy as np
np.dtypes.StringDType
""")
print(next(node.infer()))
```

### Current behavior

Raises `InferenceError`. The same holds for every other class in `numpy.dtypes` (e.g. `np.dtypes.BoolDType`), on all supported NumPy versions. The classes are attached to the module dynamically at import time, so static analysis never sees them.

### Expected behavior

`np.dtypes.StringDType` (and the other DType classes) infer as classes. `StringDType` was a headline NumPy 2.0 feature, so pylint currently reports false positives on code using it.

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output

4.4.0-dev0
